src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -165,155 +263,714 @@ private MessageConsumer newReplyStream(
+                .put(begin.buffer(), offsetBeginFrame, begin.sizeof())
                 .build();
 
-            receiver.accept(reset.typeId(), reset.buffer(), reset.offset(), reset.sizeof());
+            doNetworkData(traceId, authorization, 0L, beginWithFrameHeader);
         }
 
-        private void doWindow(
+        private void doEncodeAttach(
             long traceId,
-            int initialCredit)
+            long authorization,
+            AmqpRole role,
+            AmqpSenderSettleMode senderSettleMode,
+            AmqpReceiverSettleMode receiverSettleMode,
+            String address)
         {
-            if (initialCredit > 0)
+            int offsetAttachFrame = amqpFrameHeaderRW.limit();
+
+            AmqpAttachFW.Builder attachRW = amqpAttachRW.wrap(frameBuffer, offsetAttachFrame, frameBuffer.capacity())
+                .name(asStringFW(linkName))
+                .handle(handle)
+                .role(role)
+                .sndSettleMode(senderSettleMode)
+                .rcvSettleMode(receiverSettleMode);
+            if (role == AmqpRole.SENDER)
+            {
+                AmqpSourceListFW sourceList = new AmqpSourceListFW.Builder()
+                    .wrap(terminusListBuffer, 0, terminusListBuffer.capacity())
+                    .address(asStringFW(address))
+                    .build();
+                attachRW.source(b -> b.sourceList(sourceList));
+            }
+            else if (role == AmqpRole.RECEIVER)
             {
-                initialBudget += initialCredit;
+                AmqpTargetListFW targetList = new AmqpTargetListFW.Builder()
+                    .wrap(terminusListBuffer, 0, terminusListBuffer.capacity())
+                    .address(asStringFW(address))
+                    .build();
+                attachRW.target(b -> b.targetList(targetList));
+            }
 
-                final WindowFW window = windowRW.wrap(writeBuffer, 0, writeBuffer.capacity())
-                    .routeId(routeId)
-                    .streamId(initialId)
-                    .traceId(traceId)
-                    .budgetId(0L)
-                    .credit(initialCredit)
-                    .padding(initialPadding)
+            final AmqpAttachFW attach = attachRW.build();
+
+            AmqpFrameHeaderFW originalFrameHeader = amqpFrameHeaderRW.build();
+
+            final AmqpFrameHeaderFW updatedFrameHeader =
+                amqpFrameHeaderRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+                    .size(originalFrameHeader.sizeof() + attach.sizeof())
+                    .doff(originalFrameHeader.doff())
+                    .type(originalFrameHeader.type())
+                    .channel(originalFrameHeader.channel())
+                    .performative(b -> b.domainId(d -> d.set(originalFrameHeader.performative().domainId()))
+                        .descriptorId(d2 -> d2.set(originalFrameHeader.performative().descriptorId().get())))
```

Comment:
I have a lot of this style code in `kafka` nukleus as well.

Do you think we could modify the maven plugin to support typed non-primitive setters?
```
.channel(originalFrameHeader.channel())
.performative(originalFrameHeader.performative())
```
The resulting code above would be much simpler and more efficient.

Reply:
The possible design for the non-premitive setters for `struct` builder:
```
// AmqpFrameHeaderFW.Builder
public Builder performative(
    PerformativeFW value)
{
    assert lastFieldSet == INDEX_PERFORMATIVE - 1;
    PerformativeFW.Builder performativeRW = this.performativeRW.wrap(buffer(), limit(), maxLimit());
    performativeRW.fields(value.buffer(), value.offset(), value.sizeof());
    limit(performativeRW.build().limit());
    lastFieldSet = INDEX_PERFORMATIVE;
    return this;
}
```
```
// PerformativeFW.Builder
public Builder fields(
    DirectBuffer buffer,
    int index,
    int length)
{
    int newLimit = limit() + length;
    checkLimit(newLimit, maxLimit());
    buffer().putBytes(limit(), buffer, index, length);
    limit(newLimit);
    lastFieldSet = INDEX_DESCRIPTOR_ID;
    return this;
}
```

So we also need to add additional `fields` method in `struct` builder. 

Should I implement these in `nukleus-maven-plugin`?

Reply:
Discussed and resolved on slack.